packages/file-store/configstores/FileConfigstore.ts

```diff
@@ -0,0 +1,44 @@
+import fs from 'node:fs/promises'
+import {Upload} from '@tus/server'
+
+import {Configstore} from './Types'
+
+export class FileConfigstore implements Configstore {
```

Comment:
This class does not seem to be safe for concurrent access because the underlying file system operations are not atomic. For example, a concurrent call to `list` and `delete` can return incomplete data. It's probably not the end of the world, but worth mentioning here.

Reply:
Good point. `list` is only called in a CRON to clean expired uploads. That shouldn't happen often (or not at all if you don't use the extension). That leaves us with `delete` and `get`, for which I reckon that shouldn't really happen if there is only one client operating on the ID at hand.

We may be fine for now?

Reply:
In theory, most combinations of two methods here can lead to unexpected results, but the likelihood is probably low. So let's not get stuck here for now.

Reply:
This could be solved by queuing promises so that at any given time only one is running.

Reply:
Good point. We could use something like [`p-queue`](https://github.com/sindresorhus/p-queue) with `concurrency: 1`.

Reply:
That might already be sufficient, yes.